### DISCUSSION TITLE
Topic regex patterns and partitioned topics

### DISCUSSION THREAD

**Discussion by thomas-turner** [apache/pulsar#20716] [Q&A] [answered]
Hey,

Has something changed in Pulser 2.10+ where you now need to specify the `partition-x` in topic patterns? Previously we had patterns like `public/namespace/message_type-\w{8}` (8 char id in topic name) which in Pulsar 2.9.x worked fine for us. Try to update to the 3.0.0 and it appears like the client is not picking up any of our topics. Changing it to `public/namespace/message_type-\\w{8}-partition-\\d` results in the topics being found however I don't really want to have to include the partition in the pattern.

Thanks

**Comment by tisonkun** [comment]
@thomas-turner are your topics persistent or non-persistent?

**Reply by thomas-turner** [reply]
persistent

**Reply by coderzc** [reply, marked as answer]
https://github.com/apache/pulsar/pull/14804 introduced the feature of filtering on the broker side, but it will filter the topic list with the original contains the partition. You can set `enableBrokerSideSubscriptionPatternEvaluation=false` in the broker.conf to disable this feature.
